### PR TITLE
chore: report name changed

### LINKED ISSUE #268: Change naming schema of generated conformance tests report
## Problem Statement

In GWAPI the issue

- https://github.com/kubernetes-sigs/gateway-api/issues/2740 

has been implemented. Thus generated report should match the naming convention and directory structure. See e.g.

https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.1.0/istio-istio 

## Acceptance Criteria
- [ ] Generated report name follows new convention

### PR DESCRIPTION
**Pull request by mlavacca**
**What this PR does / why we need it**:

**Which issue this PR fixes**

Fixes #268 

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect significant changes

### CODE REVIEW

File: test/conformance/conformance_test.go [hunk 1]
```diff
@@ -87,7 +87,7 @@ func TestGatewayConformance(t *testing.T) {
 	// Currently mode only relies on the KongRouterFlavor, but in the future
 	// we may want to add more modes.
 	mode := string(config.KongRouterFlavor)
-	reportFileName := fmt.Sprintf("kong-gateway-operator-%s.yaml", mode) // TODO: https://github.com/Kong/gateway-operator/issues/268
+	reportFileName := fmt.Sprintf("standard-%s-%s-report.yaml", metadata.Release, mode)
```
Comment (pmalek) [inline test/conformance/conformance_test.go:90]: Looking at upstream reports e.g. https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports/v1.1.0/istio-istio I can see that they have the following format `${version}-report.yaml`. Why include the `standard` here?

Comment (mlavacca) [reply test/conformance/conformance_test.go:90]: this one uploaded by istio is not properly named. Here are the naming rules: https://github.com/kubernetes-sigs/gateway-api/tree/main/conformance/reports

Comment (pmalek) [reply test/conformance/conformance_test.go:90]: 😱 

Ok, good to know. 👍

Comment (mlavacca) [reply test/conformance/conformance_test.go:90]: I've submitted a PR upstream to fix it https://github.com/kubernetes-sigs/gateway-api/pull/3148

**Comment by programmer04** [comment]
`.gitignore` needs an adjustment @mlavacca

**Comment by mlavacca** [comment]
> `.gitignore` needs an adjustment @mlavacca

good catch, @programmer04! Fixed

**Review by programmer04 — APPROVED** [review]
LGTM

**Review by pmalek — APPROVED** [review]

**Pull request merged**